test/jdk/javax/rmi/ssl/SSLSocketParametersTest.java

```diff
@@ -89,117 +73,66 @@ public void runClient(Remote stub) throws IOException {
         }
     }
 
-    public static class ClientFactory extends SslRMIClientSocketFactory {
-
-        public ClientFactory() {
-            super();
-        }
-
-        public Socket createSocket(String host, int port) throws IOException {
-            System.out.println("ClientFactory::Calling createSocket(" +
-                               host + "," + port + ")");
-            return super.createSocket(host, port);
-        }
-    }
-
-    public static class ServerFactory extends SslRMIServerSocketFactory {
-
-        public ServerFactory() {
-            super();
-        }
-
-        public ServerFactory(String[] ciphers,
-                             String[] protocols,
-                             boolean need) {
-            super(ciphers, protocols, need);
-        }
-
-        public ServerFactory(SSLContext context,
-                             String[] ciphers,
-                             String[] protocols,
-                             boolean need) {
-            super(context, ciphers, protocols, need);
-        }
-
-        public ServerSocket createServerSocket(int port) throws IOException {
-            System.out.println("ServerFactory::Calling createServerSocket(" +
-                               port + ")");
-            return super.createServerSocket(port);
-        }
-    }
-
-    public void testRmiCommunication(RMIServerSocketFactory serverFactory, boolean expectException) {
-
-        HelloImpl server = null;
-        try {
-            server = new HelloImpl(0,
-                                    new ClientFactory(),
-                                    serverFactory);
-            Remote stub = server.runServer();
+    public void testRmiCommunication(RMIServerSocketFactory serverFactory) throws Exception {
+            Hello stub = (Hello)UnicastRemoteObject.exportObject(new HelloImpl(),
```

Comment:
by not retaining an explicit reference to the test rmi server, you are exposing it to potentially being GCed during the test execution  and potemntially prior to client invocation... this might sound fanciful but  this has been observed in a few scenarios due to the GC  changes ... althought it doesn't seem to have been an issue, the structure of the test appears to be inherently racy, with the potential for the client invocation to get ahead of the client with the rmi server launching bacground threads.
Any caution on not retaining a sever reference and GC interference.

Reply:
yes, good advice on reachability fence ... added some -Xcomp args to test runs and no failures detected